From a2849a0b36041f4ba7584a30a00758aaadf13014 Mon Sep 17 00:00:00 2001
From: zhengq <qi.zheng@intel.com>
Date: Fri, 3 Apr 2015 22:00:52 +0800
Subject: [PATCH] GPIO: set flag IRQCHIP_SKIP_SET_WAKE.

The irq_set_wake is not implemented in crystalcove_irqchip.
To prevent unbanlance wanring on resuming from S3 as below,
set flag IRQCHIP_SKIP_SET_WAKE.

WARNING: ... at ../kernel/irq/manage.c:524 irq_set_irq_wake+0xb3/0x100()
Unbalanced IRQ 473 wake disable

Change-Id: I1c8fc953302e199a4a7d5eabe76638deff1e9539
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-28841
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1344
Signed-off-by: zhengq <qi.zheng@intel.com>
---
 drivers/gpio/gpio-crystalcove.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpio/gpio-crystalcove.c b/drivers/gpio/gpio-crystalcove.c
index d672547..9dff39f 100644
--- a/drivers/gpio/gpio-crystalcove.c
+++ b/drivers/gpio/gpio-crystalcove.c
@@ -177,6 +177,7 @@ static struct irq_chip crystalcove_irqchip = {
 	.irq_set_type		= crystalcove_irq_type,
 	.irq_bus_lock		= crystalcove_bus_lock,
 	.irq_bus_sync_unlock	= crystalcove_bus_sync_unlock,
+	.flags = IRQCHIP_SKIP_SET_WAKE,
 };
 
 static irqreturn_t crystalcove_gpio_irq_handler(int irq, void *data)
-- 
1.9.1

